### PR TITLE
Multi-engine renaming

### LINKED ISSUE #296: Rename engine types and names used in Aviary
### Desired capability or behavior.

Rename different engine types throughout Aviary to reflect the homogenous and hetero nomenclature we developed; focus on docs and other user-facing places first, then the internal code.

### Is your feature request related to a problem? Please describe.

_No response_

### Associated Bug Report

_No response_

### LINKED ISSUE #344: pkg_resources has been deprecated
### Description

pkg_resources has been deprecated in favor of [importlib.resources](https://docs.python.org/3.11/library/importlib.resources.html#module-importlib.resources)

### Example

running `import pkg_resources` results in:
`DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html`

### Aviary Version

0.9.4-dev

### Relevant environment information

_No response_

### PR DESCRIPTION
**Pull request by johnjasa**
### Summary

Renaming any use of `multi-engine` to use homogenous or heterogeneous as appropriate.

### Related Issues

- Resolves #296 
- Resolves #344

### Backwards incompatibilities

None

### New Dependencies

None

### CODE REVIEW

File: aviary/docs/misc_resources/glossary.md [hunk 1]
```diff
@@ -39,6 +39,17 @@ design variable
 flight envelope
     The limits of speed, altitude, and load factor that an aircraft can safely operate within. Understanding the flight envelope is critical for aircraft design and operation.
 
+homogeneous engines
+    Engines that are assumed to have the same performance characteristics.
```
Comment (jkirk5) [inline aviary/docs/misc_resources/glossary.md:43]:
```suggestion
    An individual engine or set of engines that all have identical performance characteristics.
```

File: aviary/docs/misc_resources/glossary.md [hunk 2]
```diff
@@ -39,6 +39,17 @@ design variable
...
+    E.g. you might have four engines on an aircraft, but they are all assumed to have the same thrust, fuel consumption, etc.
+    Each of these engines would use the same `EngineModel` object in Aviary.
+    See also; heterogeneous engines.
+
+heterogeneous engines
+    Engines that are assumed to have different performance characteristics.
```
Comment (jkirk5) [inline aviary/docs/misc_resources/glossary.md:49]:
```suggestion
    Multiple sets of homogenous engines, with each set having unique performance characteristics.
```

File: aviary/docs/misc_resources/glossary.md [hunk 3]
```diff
@@ -39,6 +39,17 @@ design variable
...
+    This is useful if you are modeling an aircraft with engines of different sizes or types.
```
Comment (jkirk5) [inline aviary/docs/misc_resources/glossary.md:50]:
```suggestion
    This is useful if you are modeling an aircraft with engines of different sizes or types.
    Each unique engine would use the same `EngineModel` object in Aviary.
```

File: aviary/mission/flight_phase_builder.py [hunk 4]
```diff
@@ -175,7 +175,7 @@ def build_phase(self, aviary_options: AviaryValues = None, phase_type=EquationsO
             rate_targets = ['dh_dr']
             rate2_targets = ['d2h_dr2']
 
-        # For multi-engine cases, we may have throttle allocation control.
+        # For multiple heterogeneous-engine cases, we may have throttle allocation control.
```
Comment (jkirk5) [inline aviary/mission/flight_phase_builder.py:178]:
```suggestion
        # For heterogeneous-engine cases, we may have throttle allocation control.
```

File: aviary/subsystems/aerodynamics/flops_based/skin_friction.py [hunk 5]
```diff
@@ -47,7 +47,7 @@ def setup(self):
             Aircraft.VerticalTail.NUM_TAILS, zero_count)
         num_fuselages, _ = aviary_options.get_item(
             Aircraft.Fuselage.NUM_FUSELAGES, zero_count)
-        # TODO does not used vectorized multi-engine. Temp using single engine
+        # TODO does not used vectorized multiple heterogeneous engines. Temp using single engine
```
Comment (jkirk5) [inline aviary/subsystems/aerodynamics/flops_based/skin_friction.py:50]:
```suggestion
        # TODO does not used vectorized heterogeneous engines. Temp using single engine
```

File: aviary/mission/flight_phase_builder.py (hunk 4 above)
Comment (jkirk5) [reply aviary/mission/flight_phase_builder.py:178]: Implied an aircraft with heterogeneous engines must have more than one type of engine (can't have single heterogeneous engine)

File: aviary/subsystems/mass/flops_based/landing_gear.py [hunk 6]
```diff
@@ -299,7 +299,7 @@ def setup_partials(self):
 
     def compute(self, inputs, outputs, discrete_inputs=None, discrete_outputs=None):
         options: AviaryValues = self.options['aviary_options']
-        # TODO temp using first engine, multi-engine not supported
+        # TODO temp using first engine, multiple heterogeneous engines not supported
```
Comment (jkirk5) [inline aviary/subsystems/mass/flops_based/landing_gear.py:302]:
```suggestion
        # TODO temp using first engine, heterogeneous engines not supported
```

File: aviary/subsystems/mass/flops_based/landing_gear.py [hunk 7]
```diff
@@ -339,7 +339,7 @@ def compute(self, inputs, outputs, discrete_inputs=None, discrete_outputs=None):
 
     def compute_partials(self, inputs, partials, discrete_inputs=None):
         options: AviaryValues = self.options['aviary_options']
-        # TODO temp using first engine, multi-engine not supported
+        # TODO temp using first engine, multiple heterogeneous engines not supported
```
Comment (jkirk5) [inline aviary/subsystems/mass/flops_based/landing_gear.py:342]:
```suggestion
        # TODO temp using first engine, heterogeneous engines not supported
```

File: aviary/subsystems/propulsion/utils.py [hunk 8]
```diff
@@ -158,7 +158,7 @@ def build_engine_deck(aviary_options: AviaryValues, meta_data=_MetaData):
                     expected_dim = default_value.ndim
                     val_dim = aviary_val.ndim
                     # if aviary_values has one more dimension than expected per-engine,
-                    # we know aviary_values is for multi-engine type. Currently only using
+                    # we know aviary_values is for multiple heterogeneous engine type. Currently only using
```
Comment (jkirk5) [inline aviary/subsystems/propulsion/utils.py:161]:
```suggestion
                    # we know aviary_values is for heterogeneous engine type. Currently only using
```

File: aviary/utils/preprocessors.py [hunk 9]
```diff
@@ -114,8 +114,8 @@ def preprocess_propulsion(aviary_options: AviaryValues, engine_models: list = No
     aviary_options, an EngineDeck is created using avaliable inputs and options in
     aviary_options.
 
-    Vectorizes variables in aviary_options in the correct order for multi-engine
-    vehicles.
+    Vectorizes variables in aviary_options in the correct order for vehicles with
+    multiple heterogeneous engines.
```
Comment (jkirk5) [inline aviary/utils/preprocessors.py:118]:
```suggestion
    heterogeneous engines.
```

File: aviary/utils/process_input_decks.py [hunk 10]
```diff
@@ -378,7 +378,7 @@ def initial_guessing(aircraft_values: AviaryValues, initial_guesses, engine_buil
         total_thrust = aircraft_values.get_val(
             Aircraft.Engine.SCALED_SLS_THRUST, 'lbf') * aircraft_values.get_val(Aircraft.Engine.NUM_ENGINES)
     except KeyError:
-        # Multi-engine-model case. Get thrust from the engine decks instead.
+        # multiple heterogeneous engine-model case. Get thrust from the engine decks instead.
```
Comment (jkirk5) [inline aviary/utils/process_input_decks.py:381]:
```suggestion
        # heterogeneous engine-model case. Get thrust from the engine decks instead.
```

File: aviary/variable_info/variable_meta_data.py [hunk 11]
```diff
@@ -4263,7 +4263,7 @@
     desc='Total propulsion group mass'
 )
 
-# TODO clash with per-engine scaling, need to resolve w/ multi-engine
+# TODO clash with per-engine scaling, need to resolve w/ multiple heterogeneous engine
```
Comment (jkirk5) [inline aviary/variable_info/variable_meta_data.py:4266]:
```suggestion
# TODO clash with per-engine scaling, need to resolve w/ heterogeneous engine
```

File: aviary/subsystems/mass/flops_based/misc_engine.py [hunk 12]
```diff
@@ -16,7 +16,7 @@ class EngineMiscMass(om.ExplicitComponent):
     Calculates total sum of all misc engine mass on the aircraft
 
     Currently using engine-level additional mass (scaled by num_engines)
-    and propulsion-level starter and controls mass, not multi-engine safe!
+    and propulsion-level starter and controls mass, not multiple heterogeneous engines safe!
```
Comment (jkirk5) [inline aviary/subsystems/mass/flops_based/misc_engine.py:19]:
```suggestion
    and propulsion-level starter and controls mass, not heterogeneous engine safe!
```

File: aviary/docs/misc_resources/comparison_to_flops.md [hunk 13]
```diff
@@ -93,7 +93,8 @@ complex than for a traditional tube-and-wing configuration.
 
 FLOPS allows for the input of a maximum of two engine types, each with its own
 engine deck, whereas in Aviary an unlimited number of engine types may be
-defined. Aviary retains many of the relations used by FLOPS to scale engine mass and
+defined. This means that Aviary supports arbitrary numbers of heterogenous
+engines Aviary retains many of the relations used by FLOPS to scale engine mass and
```
Comment (jkirk5) [inline aviary/docs/misc_resources/comparison_to_flops.md:97]:
```suggestion
engines, each of which has unique characteristics and can use different methodologies for sizing and performance estimation. Aviary retains many of the relations used by FLOPS to scale engine mass and
```

**Review by jkirk5 — CHANGES_REQUESTED** [review]

File: aviary/docs/misc_resources/glossary.md [hunk 14]
```diff
@@ -39,6 +39,17 @@ design variable
 flight envelope
     The limits of speed, altitude, and load factor that an aircraft can safely operate within. Understanding the flight envelope is critical for aircraft design and operation.
 
+homogeneous engines
+    Engines that are assumed to have the same performance characteristics.
+    E.g. you might have four engines on an aircraft, but they are all assumed to have the same thrust, fuel consumption, etc.
+    Each of these engines would use the same `EngineModel` object in Aviary.
```
Comment (jkirk5) [inline aviary/docs/misc_resources/glossary.md:45]:
```suggestion
    All engines in a homogenous set use the same `EngineModel` object in Aviary.
```

**Review by Kenneth-T-Moore — APPROVED** [review]
Fix works great.

**Review by jkirk5 — APPROVED** [review]

**Comment by xjjiang** [comment]
@johnjasa there is usage of  pkg_resources in `docs/user_guide/external_aero.ipynb`

**Pull request merged**